grpclog/component.go

```diff
@@ -28,57 +28,58 @@ import (
 	"google.golang.org/grpc/internal/grpclog"
 )
 
-// Component is a unique identifier for a component. Created with NewComponent.
-type Component string
-
-// componentData records the settings for a component.
-type componentData struct {
+// ComponentData records the settings for a component.
+type ComponentData struct {
+	name      string
 	verbosity int
 	level     int
 }
 
 const envName = "GRPC_GO_LOG_LEVEL"
 const sentinel = math.MinInt32
 
-// grpc components
+// log levels
 const (
-	CORE       = Component("CORE")
-	ALTS       = Component("ALTS")
-	GRPCLB     = Component("BALANCER_GRPCLB")
-	RLS        = Component("BALANCER_RLS")
-	ROUNDROBIN = Component("BALANCER_ROUNDROBIN")
-	BINARYLOG  = Component("BINARYLOG")
-	CHANNELZ   = Component("CHANNELZ")
-	DNS        = Component("RESOLVER_DNS")
-	TRANSPORT  = Component("TRANSPORT")
-	XDS        = Component("XDS")
+	levelInfo    = 0
+	levelWarning = 1
+	levelError   = 2
+)
+
+// grpc components
+var (
+	Core       = ComponentData{"CORE", 0, 0}
+	Alts       = ComponentData{"ALTS", 0, 0}
+	Grpclb     = ComponentData{"BALANCER_GRPCLB", 0, 0}
+	Rls        = ComponentData{"BALANCER_RLS", 0, 0}
+	RoundRobin = ComponentData{"BALANCER_ROUNDROBIN", 0, 0}
+	BinaryLog  = ComponentData{"BINARYLOG", 0, 0}
+	Channelz   = ComponentData{"CHANNELZ", 0, 0}
+	DNS        = ComponentData{"RESOLVER_DNS", 0, 0}
+	Transport  = ComponentData{"TRANSPORT", 0, 0}
+	Xds        = ComponentData{"XDS", 0, 0}
```

Comment:
XDS

Reply:
This has been replaced with an in-package defined "logger" var.